### ISSUE TITLE
Problems with 1+ and list*

### ISSUE THREAD

**Issue by drmeister** [drmeister/clasp#167]
16:21 <stassats`> drmeister: (1+ 1 2 3) => 2
16:23 <stassats`> drmeister: (list*) => NIL

**Comment by drmeister** [comment]
This happens because cclasp takes expressions entered at the REPL and passes them to clasp-cleavir::cclasp-eval-with-env and it in turn does this to simple expressions:
               (apply name (loop for arg in body
                              collect (cclasp-eval arg env)))

For some reason the lambda list processing of the applied function is not recognizing that too many arguments are being provided.
Everything else seems to work fine though:
(apply (compile nil '(lambda (x) (+ 1 x))) (list 2 3 4)) --> 3
(funcall (compile nil '(lambda (x) (+ 1 x))) 2 3 4) --> 3
(funcall (compile nil '(lambda (x y) (+ y x))) 2 3 4) --> 5

**Comment by drmeister** [comment]
This has been fixed in 565c5a22870aeeb31e75da07f1cb39ca7203de81  - I forgot to add the issue number to the commit message

**Comment by Shinmera** [comment]
On 3b83c17c666a05ab4f23aac5d13335edff463aec
```
> (1+ 1 2 3)

Condition of type: SIMPLE-PROGRAM-ERROR
Too many arguments for cl->1+ - got 3 and expected 1

Available restarts:
(use :r1 to invoke restart 1)

1. (RESTART-TOPLEVEL) Go back to Top-Level REPL.

Broken at frame[116] TOP-LEVEL.
 File: #P"/media/autobuild/linut/clasp-testing/.autobuild/3b83c17c666a05ab4f23aac5d13335edff463aec/build/clasp/Contents/Resources/lisp/kernel/lsp/top.lsp" (Position #21029)
>> :r1

> (1+ 1)

2
> (list*)

NIL
```

**Comment by Shinmera** [comment]
Fixed on 6d0d3683128f0ae753b63680a9f72d86da5becd5

**Comment by drmeister** [comment]
The error reported by (list*) is the wrong kind

**Comment by drmeister** [comment]
This should all be fixed now

**Comment by Shinmera** [comment]
Should I get some error message print or is the debugger just shite and this is correct?
```
> (list*)

Condition of type: ARGUMENT-NUMBER-ERROR
#<CORE:ARGUMENT-NUMBER-ERROR @0x2b0ef8bc06f1>

Available restarts:
(use :r1 to invoke restart 1)
```

**Comment by drmeister** [comment]
I added a report to argument-number-error - it looks nice now.

**Comment by Shinmera** [comment]
Fixed on 8c9e60ecd2759c10f490fda28cdfe5f7889eb52b
```
> (list*)

Condition of type: ARGUMENT-NUMBER-ERROR
At least 1 argument required, 0 arguments supplied.
```
